### ISSUE TITLE
Clear buttons inside fields are not accessible via keyboard

### ISSUE THREAD

**Issue by claudiahub** [backstage/backstage#11292]
## Expected Behavior

Clear button should be accessible through keyboard for 'Owner', 'Lifecycle', 'Tags'.

## Actual Behavior

1. Navigate to `Home` on the left navigation. 
2. Navigate to the main section of the page and launch NVDA.
3. Verify after inputing some data in `Owner` or `Lifecycle` or `Tags` that the clear button in those fields are accessible through keyboard.

<img width="1329" alt="image" src="https://user-images.githubusercontent.com/54085737/166659517-556f00ab-8804-4da2-babd-7b065b63ae43.png">

## Context

Ensure all interactive elements can receive focus and are actionable via keyboard. For example, links use the ENTER key while buttons should allow ENTER and SPACE key.
Clear button on top of each option in combo box is also not accessible.

Involved component(s):

- EntityLifecyclePicker from '@backstage/plugin-catalog-react'
- EntityOwnerPicker from '@backstage/plugin-catalog-react'
- EntityTypePicker from '@backstage/plugin-catalog-react'
- UserListPicker from '@backstage/plugin-catalog-react'

User Impact: 
When any UI component are not accessible by using keyboard then user who rely on keyboard are facing difficulty and not able to perform any action and are completely block to perform any functionality on that control element.

A11y Conformance Level: A

WCAG Success Criteria: 2.1.1

## Environment

Browser Information: 
Google Chrome v100.0.4896.127
Firefox 99.0.1
Safari 15.4

**Comment by claudiahub** [comment]
It seems clear buttons, placed within `Autocomplete` comboboxes, display on mouse hover only.
For this reason, this component shouldn't be triggered by keyboard actions.
The autocomplete follows the https://www.w3.org/TR/wai-aria-practices/#combobox spec.

Reference to Material UI https://github.com/mui/material-ui/issues/30815

**Comment by Rugvip** [comment]
Got a bit confused here, but your second comment is highlighting the same thing as I conclude as well right, that this is actually the intended behavior? Is it fine to close this issue?

**Comment by claudiahub** [comment]
@Rugvip sure, it is fine to close this issue since the component it's located on the materialUI side. thanks!
